cluster/addons/calico-policy-controller/calico-clusterrole.yaml

```diff
@@ -36,6 +36,7 @@ rules:
       - get
       - list
       - watch
+      - patch
```

Comment:
write access to pods is new, and atypical for a network plugin. why does calico require this?

Reply:
When using the Kubernetes datastore driver (i.e. no extra database for Calico), the Calico CNI plugin writes the IP allocation back to the Pod as an annotation.  Without that, we had to wait for the kubelet to write the IP back to the Pod status, which can take 30s.

Reply:
that can be done with an update of pods/status, which is more appropriate, since it doesn't let calico modify the pod spec (including the image being run)

Reply:
We initially had a solution that patched the PodIP field but we found that kubelet will often back out that change and then reapply it later so the fix didn't work.

Reply:
> the Calico CNI plugin writes the IP allocation back to the Pod as an annotation

the annotation can be written as part of a status patch

Reply:
Are you sure about that?  I thought we'd tried that and it didn't work.

Reply:
yes

```sh
curl http://localhost:8080/api/v1/namespaces/default/pods/busybox/status \
  -X PATCH \
  -H "Content-Type:application/merge-patch+json" \
  --data '{"metadata":{"annotations":{"test":"value"}}}'

{
  "kind": "Pod",
  "apiVersion": "v1",
  "metadata": {
    "name": "busybox",
    "namespace": "default",
    "selfLink": "/api/v1/namespaces/default/pods/busybox/status",
    "uid": "c3dceff8-6f22-11e8-a4aa-acbc32c1ca87",
    "resourceVersion": "2550",
    "creationTimestamp": "2018-06-13T15:59:34Z",
    "annotations": {
      "test": "value"
    }
  },
...
```

Reply:
I've raised an issue against Calico to track this - agree it would be nice to clean up. https://github.com/projectcalico/libcalico-go/issues/888